### PR TITLE
Migrate removed networking variables

### PR DESCRIPTION
**Pull request by joe4dev**
* Remove/migrate edge variables EDGE_PORT/EDGE_PORT_HTTP/EDGE_BIND_HOST. All write-usages. Few read-usages remain until the respective integration tools are updated to adopt `AWS_ENDPOINT_URL`
* Un-deprecate `USE_SSL` and describe its current usage
* Remove `<SERVICE>_PORT_EXTERNAL`

### CODE REVIEW

File: content/en/references/configuration.md
Comment (simonrw) [inline content/en/references/configuration.md:1]: `EDGE_FORWARD_URL` is also not used, perhaps we should remove usage of it?

**Review by simonrw — CHANGES_REQUESTED** [review]
We need to update usages of e.g. `HOSTNAME_EXTERNAL` and other deprecated variables from the main text as well as the configuration section. There are a few references to `HOSTNAME_EXTERNAL` but we need to replace `LOCALSTACK_HOSTNAME` as well (unless used in a read-only capacity).

In addition, could we make a change to the FAQ entry about "Why can’t my other Docker containers reach LocalStack?"? Perhaps keep the first paragraph, but remove the remaining text and link to the network troubleshooting guide?

File: content/en/references/configuration.md
Comment (joe4dev) [reply content/en/references/configuration.md:1]: yeah, I was wondering whether that belongs to the legacy edge proxy but let's just remove it in the same go 👍

**Comment by joe4dev** [comment]
> We need to update usages of e.g. `HOSTNAME_EXTERNAL` and other deprecated variables from the main text as well as the configuration section. There are a few references to `HOSTNAME_EXTERNAL` but we need to replace `LOCALSTACK_HOSTNAME` as well (unless used in a read-only capacity).


Correct, that's quite some usage.
I replace them except for transparent endpoint injection (done in the Lambda PR https://github.com/localstack/docs/pull/899) and LocalStack tools that do not yet support `AWS_ENDPOINT_URL` (I added them to the task list in the decision doc).

> In addition, could we make a change to the FAQ entry about "Why can’t my other Docker containers reach LocalStack?"? Perhaps keep the first paragraph, but remove the remaining text and link to the network troubleshooting guide?

Good suggestion, done 👍

File: content/en/user-guide/aws/elastic-kubernetes-service/index.md [hunk 1]
```diff
@@ -100,12 +100,13 @@ You can now use ECR (Elastic Container Registry) images within your EKS environm
 
 #### Initial configuration
 
-To modify the return value of resource URIs for most services, including ECR, you can utilize the `HOSTNAME_EXTERNAL` variable in the [configuration]({{< ref "configuration" >}}). By default, ECR returns a `repositoryUri` starting with `localhost`, such as: `localhost:<port>/<repository-name>`. However, if you set the `HOSTNAME_EXTERNAL` to `localhost.localstack.cloud`, the ECR will provide a `repositoryUri` like `localhost.localstack.cloud:<port>/<repository_name>`.
+To modify the return value of resource URIs for most services, including ECR, you can utilize the `LOCALSTACK_HOST` variable in the [configuration]({{< ref "configuration" >}}).
+By default, ECR returns a `repositoryUri` starting with `localhost.localstack.cloud`, such as: `localhost.localstack.cloud:<port>/<repository-name>`.
```
Comment (joe4dev) [inline content/en/user-guide/aws/elastic-kubernetes-service/index.md:104]: @simonrw That's the new default, right?

File: content/en/user-guide/ci/gitlab-ci/index.md [hunk 2]
```diff
@@ -62,6 +60,13 @@ test:
     - aws --endpoint http://localhost.localstack.cloud:4566 s3 ls s3://test/
 ```
 
+<details>
+<summary>For LocalStack versions before 3.0.0</summary>
```
Comment (joe4dev) [inline content/en/user-guide/ci/gitlab-ci/index.md:64]: looks a bit awkward but probably needed

File: content/en/user-guide/aws/elastic-kubernetes-service/index.md (hunk 1 above)
Comment (simonrw) [reply content/en/user-guide/aws/elastic-kubernetes-service/index.md:104]: Yes this is correct

**Review by simonrw — APPROVED** [review]
Looks great, thanks a lot for taking care of this migration. It's exciting to be able to remove `HOSTNAME_EXTERNAL` and `LOCALSTACK_HOSTNAME` from our docs!

**Pull request merged**